kubetest/e2e.go

```diff
@@ -725,3 +731,26 @@ func toBuildTesterOptions(o *options) *e2e.BuildTesterOptions {
 		Parallelism: o.ginkgoParallel.Get(),
 	}
 }
+
+func setKubeShhBastionEnv(o *options) error {
+	value, err := control.Output(exec.Command(
+		"gcloud", "compute", "instances", "describe",
```

Comment:
This makes the feature gce/gke-specific. I'm fine with that but:
- we probably should add some validation that this option can be used only with gce/gke providers
- I would like someone from test-infra team to also take a look.
@BenTheElder @cjwagner ?

Reply:
kubetest/ is a giant mess, there's slow progress towards replacing it.

You'll need to assign an actual approver for this change :sweat_smile: (https://github.com/kubernetes/test-infra/pull/11813)

/unassign

Reply:
Given that there is an effort to rewrite `kubetest` I'm disinclined to block too hard on this, but a good way to denote it as provider-specific would be to do this in the `TestSetup()` implementation of the relevant deployers (`bash` with a gate on `o.Provider`, and `gke` would be my guess)

Reply:
Done.